code/controllers/subsystem/processing/networks.dm

```diff
@@ -1,51 +1,340 @@
-PROCESSING_SUBSYSTEM_DEF(networks)
+#define DEBUG_NETWORKS
+
+
+SUBSYSTEM_DEF(networks)
 	name = "Networks"
 	priority = FIRE_PRIORITY_NETWORKS
-	wait = 1
-	stat_tag = "NET"
+	wait = 5
 	flags = SS_KEEP_TIMING
 	init_order = INIT_ORDER_NETWORKS
+
+	var/list/relays = list()
 	var/datum/ntnet/station/station_network
-	var/assignment_hardware_id = HID_RESTRICTED_END
-	var/list/networks_by_id = list()				//id = network
-	var/list/interfaces_by_id = list()				//hardware id = component interface
-	var/resolve_collisions = TRUE
+	var/list/network_initialize_queue = list()
+	// all interfaces by their hardware address.
+	var/list/interfaces_by_hardware_id = list()
+
+	// Fire information
+	var/list/networks = list()
+	// Why not list?  Because its a Copy() every time we add a packet, and thats stupid.
+	var/datum/netdata/first = null // start of the queue.  Pulled off in fire.
+	var/datum/netdata/last = null	// end of the queue.  pushed on by transmit
+	var/packet_count = 0
+	// packet stats
+	var/count_broadcasts_packets = 0 // count of broadcast packets sent
+	var/count_failed_packets = 0 	// count of message fails
+	var/count_good_packets = 0
+	// Logs moved here
+	// Amount of logs the system tries to keep in memory. Keep below 999 to prevent byond from acting weirdly.
```

Comment:
define "acting weirdly" here

Reply:
Not sure, it was part of the old log system I cut and pasted over.